include/TileComponent.h

```diff
@@ -17,17 +19,27 @@ class TileComponent : public Component
 
 	SDL_Rect tileRect;
 	int tileID;
-	const char* path;
+	TexturesEnum texture;
 
 	TileComponent() = default;
-	TileComponent(int x, int y, int w, int h, int id, const std::map<int, std::pair<std::string, bool>>* textureDict);
+	TileComponent(int x, int y, int w, int h, int id, const std::map<int, std::pair<TexturesEnum, bool>>* textureDict);
 	~TileComponent() = default;
 
 	void init() override;
 
-	bool hasCollision(){return this->collision;}
-	std::string getName(){return this->tileName;}
+	bool hasCollision() {
+		return this->collision;
+	}
+
+	std::string getName() {
+#ifdef TEXTURE_ENUM_DEFINED
+		return std::string(magic_enum::enum_name(this->texture));
+#else
+		return "Undefined Enum";
+#endif
```

Comment:
Compiler declarations make the code less readable, avoid if possible

Reply:
This can probably be removed, as either a compiler error will be thrown, or one can be thrown using `static_assert`. Either way, please remove compiler declaration

Reply:
As discussed, this function isn't used, thus can be removed. This also means `magic_enum` is not needed.